src/dotnet/commands/dotnet-nuget/NuGetCommand.cs

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using Microsoft.DotNet.Cli.CommandLine;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.InternalAbstractions;
+using NugetProgram = NuGet.CommandLine.XPlat.Program;
+
+namespace Microsoft.DotNet.Tools.NuGet
+{
+    public class NuGetCommand
+    {
+        public static int Run(string[] args)
+        {
+            DebugHelper.HandleDebugSwitch(ref args);
+
+            var app = new CommandLineApplication(false)
+            {
+                Name = "dotnet nuget",
+                FullName = ".NET NuGet command runner",
+                Description = "For running locals, push and delete commands (\"dotnet nuget [command] --help\" for specifics)"
+            };
+
+
+            app.OnExecute(() =>
+            {
+                try
+                {
+                    return RunCommand(args, new NuGetCommandRunner());
+                }
+                catch (InvalidOperationException e)
+                {
+                    Console.WriteLine(e.Message);
+
+                    return -1;
+                }
+                catch (Exception e)
+                {
+                    Console.WriteLine(e.Message);
+
+                    return -2;
+                }
+            });
+
+            return app.Execute(args);
+        }
+
+        public static int RunCommand(IEnumerable<string> args, INuGetCommandRunner commandRunner)
+        {
+            Debug.Assert(commandRunner != null, "A command runner must be passed to RunCommand");
+            if (commandRunner == null)
+            {
+                return -1; // invalid operation
+            }
+
+            string[] commandArgs = args.ToArray();
+            if (commandArgs.Length == 0 ||
+                !commandArgs[0].Equals("locals", StringComparison.OrdinalIgnoreCase) &&
+                !commandArgs[0].Equals("push", StringComparison.OrdinalIgnoreCase) &&
+                !commandArgs[0].Equals("delete", StringComparison.OrdinalIgnoreCase))
+            {
+                return 1; // unsupported argument
+            }
+
+            return commandRunner.Run(commandArgs);
+        }
+
+        private class NuGetCommandRunner : INuGetCommandRunner
+        {
+            public int Run(string[] commandArgs)
+            {
+                var nugetAsm = typeof(NugetProgram).GetTypeInfo().Assembly;
```

Comment:
We are doing something very similar here in the `restore` command. https://github.com/dotnet/cli/blob/rel/1.0.0/src/dotnet/commands/dotnet-restore/NuGet3.cs

Should we consolidate `restore` with the rest of the nuget commands? But it will break a lot of users (using `dotnet restore` instead of `dotnet nuget restore`). Let us at least try to consolidate at source code level and avoid duplicating code.